ecl/hqlcpp/hqlsource.cpp

```diff
@@ -2819,34 +2819,54 @@ void DiskReadBuilderBase::buildTransformFpos(BuildCtx & transformCtx)
 
 void DiskReadBuilderBase::extractMonitors(IHqlExpression * ds, SharedHqlExpr & unkeyedFilter, HqlExprArray & conds)
 {
-    ForEachItemIn(i, conds)
+    HqlExprAttr mode = tableExpr->queryChild(2);
+    //KEYED filters can only currently be implemented for binary files - not csv, xml or pipe....
```

Comment:
We would want to lift that restriction though - for remote filtering purposes. Is there a Jira to do so?

Reply:
I will create one.  It will be interesting to implement.  I think it might require our current csv and xml reading to be restructured. I suspect it would make everything cleaner in the process.
HPCC-19158